syntax_and_semantics/annotations/user_defined_annotations.md

```diff
@@ -0,0 +1,172 @@
+annotation MyClass; end
+annotation MyMethod; end
+annotation MyIvar; end
+
+@[MyClass]
+class Foo
+  {{@type.annotation(MyClass).stringify}}
+  
+  @[MyIvar]
+  @num : Int32 = 1
+  
+  @[MyIvar]
+  property name : String = "jim"
+
+  def properties
+    {% for ivar in @type.instance_vars %}
+      {{ivar.annotation(MyIvar).stringify}}
+    {% end %}
+  end
+end
+
+@[MyMethod]
+def my_method
+  {{@def.annotation(MyMethod).stringify}}
+end
+
+Foo.new.properties
+my_method
+
+# Which would print
+"@[MyClass]"
+"@[MyIvar]"
+"@[MyIvar]"
+"@[MyMethod]"
+```
+
+### Reading Multiple Annotations
+
+If there are multiple annotations of the same type applied to the same instance variable/method/type, the `annotations(type : TypeNode)` method can be used.  This will work on anything that `annotation(type : TypeNode)` would, but instead returns an `ArrayLiteral(Annotation)`.
+
+```crystal
+annotation MyAnnotation; end
+
+@[MyAnnotation("foo")]
+@[MyAnnotation(123)]
+@[MyAnnotation(123)]
+def annotation_read
+  {% for ann, idx in @def.annotations(MyAnnotation) %}
+    pp "Annotation {{idx}} = {{ann[0].id}}"
+  {% end %}
+end
+
+annotation_read
+
+# Which would print
+"Annotation 0 = foo"
+"Annotation 1 = 123"
+"Annotation 2 = 123"
+```
+
+## Fields
+
+Data can be stored within an annotation.
+
+```crystal
+annotation MyAnnotaion; end
+
+# The fields can either be a key/value pair
+@[MyAnnotation(key: "value", value: 123)]
+
+# Or index based
+@[MyAnnotation("foo", 123, false)]
+```
+
+### Key/value
+
+The values of key/value pairs would be readable at compile time via the [`[]`](https://crystal-lang.org/api/Crystal/Macros/Annotation.html#%5B%5D%28name%3ASymbolLiteral%7CStringLiteral%7CMacroId%29%3AASTNode-instance-method) method.
+
+```crystal
+annotation MyAnnotation; end
+
+@[MyAnnotation(default: 2)]
+def double(num : Number? = nil)
+  # The name can be a `String`, `Symbol`, or `MacroId`
+  2 * (num ? num : {{@def.annotation(MyAnnotation)[:default]}})
+end
+
+double 10 # => 20
+double # => 4
+```
+
+### Indexed
+
+Values added without a key name can be read at compile time via the [`[]`](<https://crystal-lang.org/api/Crystal/Macros/Annotation.html#%5B%5D%28index%3ANumberLiteral%29%3AASTNode-instance-method>) method, however only one index can be read at a time.
+
+```crystal
+annotation MyAnnotation; end
+
+@[MyAnnotation(1,2,3,4)]
+def annotation_read
+  {% for idx in [0,1,2,3,4] %}
+    {% value = @def.annotation(MyAnnotation)[idx] %}
+    pp "{{idx}} = {{value}}"
+  {% end %}
+end
+
+annotation_read
+
+# Which would print
+"0 = 1"
+"1 = 2"
+"2 = 3"
+"3 = 4"
+"4 = nil"
+```
+
+## Usages
+
+Annotations are best used to store metadata about a given instance variable, type, or method; so that it can be read at compile time using macros.  One of the main benefits of annotations is that they are applied directly to instance variables/methods. Because of this classes look more natural since a standard macro is not needed to create these properties/methods.
```

Comment:
```suggestion
Annotations are best used to store metadata about a given instance variable, type, or method so that it can be read at compile time using macros.  One of the main benefits of annotations is that they are applied directly to instance variables/methods, which causes classes to look more natural since a standard macro is not needed to create these properties/methods.
```